cdap-proto/src/main/java/co/cask/cdap/proto/Id.java

```diff
@@ -383,18 +383,17 @@ public int hashCode() {
    * Id.Stream uniquely identifies a stream.
    */
   public static final class Stream {
-    private final String namespace;
+    private final Namespace namespace;
```

Comment:
this affect the efficiency of the computation of hashCode for `Id.Stream`. You'd like to change the `Id.Namespace` hashCode computation to be efficient as well.